api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -105,15 +105,15 @@ public void validate(Object target, Errors errors) {
 		// check attributes
 		super.validateAttributes(visit, errors, Context.getVisitService().getAllVisitAttributeTypes());
 		
-		// check start and end dates
+		// check for overlapping visits
 		if (disallowOverlappingVisits()) {
```

Comment:
What exactly are you trying to achieve here?  There is no context as to what the problem is that you are trying to solve in the ticket description, and nothing here that indicates what the problem is that you are solving.  What was wrong before and how does this fix it?

Reply:
The goal here is basically efficiency (though the ticket may be poorly described). The `VisitValidator` currently works by loading _all_ non-voided visits for the patient and then checking one-by-one that the start date and stop date don't overlap with the visit being validated in Java code. This means that we hit the n+1 selects problem as well as unnecessarily materializing entire Visit objects just to check two properties.

The change here basically aims to push the work down to the database; i.e., instead of loading all visits for a given patient, we load all visits where the period between the visits start date and stop date overlaps with that of the visit being edited. If there is more than one visit or if the single visit is not the same visit as the visit being edited, then we know there's an overlapping visit and can reject the request.

Reply:
@mseaton @mogoodrich I'm really sorry about that. The PR was still a work in progress, so I hadn't included any details yet. Thanks @ibacher for explaining the issue.